### PR TITLE
Merging feature branch for Cluster Secret Service to Main

### PR DESCRIPTION
**Pull request by Shubbhang351**
1. **What this PR does / why we need it**:

Merging feature branch for Cluster Secret Service to Main

### CODE REVIEW

**Comment by ramya-bangera** [comment]
@Shubbhang351 - I do not see the build tags changes added to this PR, am I missing something?

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

File: internal/resources/kubernetessecret/spec/spec.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	"fmt"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	secretmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster"
+)
+
+//nolint:all
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/spec/spec.go]: updated PTAL

File: internal/resources/kubernetessecret/scope/scope.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+func FlattenScope(scopedFullname *ScopedFullname, scopesAllowed []string) (data []interface{}, name string) {
+	if scopedFullname == nil {
+		return data, name
+	}
+
+	flattenScopeData := make(map[string]interface{})
+
+	// nolint: all
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/scope/scope.go]: updated PTAL

File: internal/resources/kubernetessecret/resource_secret_mock_test.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+// Register a new responder when the given call is made.
+func changeStateResponder(registerFunc func(), successResponse int, successResponseBody interface{}) httpmock.Responder {
+	return func(r *http.Request) (*http.Response, error) {
+		registerFunc()
+		return httpmock.NewJsonResponse(successResponse, successResponseBody)
+	}
+}
+
+// Function to set up HTTP mocks for the kubernetes secret requests anticipated by this test, when not being run against a real TMC stack.
+// nolint: ineffassign,staticcheck
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/resource_secret_mock_test.go]: updated PTAL

File: internal/resources/kubernetessecret/constants.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package kubernetessecret
+
+//nolint:all
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/constants.go]: updated PTAL

File: internal/models/kubernetessecret/cluster/secret_spec.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+//   - SECRET_TYPE_DOCKERCONFIGJSON: SECRET_TYPE_DOCKERCONFIGJSON, Kubernetes secrets type : kubernetes.io/dockerconfigjson.
+//
+// swagger:model vmware.tanzu.manage.v1alpha1.cluster.namespace.secret.SecretType
+type VmwareTanzuManageV1alpha1ClusterNamespaceSecretType string
+
+func NewVmwareTanzuManageV1alpha1ClusterNamespaceSecretType(value VmwareTanzuManageV1alpha1ClusterNamespaceSecretType) *VmwareTanzuManageV1alpha1ClusterNamespaceSecretType {
+	return &value
+}
+
+//nolint:all
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/models/kubernetessecret/cluster/secret_spec.go]: updated PTAL

**Review by ishangupta-ds — APPROVED** [review]

**Comment by ishangupta-ds** [comment]
@Shubbhang351 Before merging please ensure the commits are proper - I see 16 commits and the commit messages are not intuitive https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/254/commits

**Review by ishangupta-ds — DISMISSED** [review]

**Review by ramya-bangera — DISMISSED** [review]

**Review by ramya-bangera — APPROVED** [review]

**Review by ishangupta-ds — APPROVED** [review]

**Pull request merged**